lib/id3c/cli/redcap.py

```diff
@@ -795,3 +802,13 @@ def __str__(self):
         # to see things exactly as they were, with minimal post-processing,
         # when troubleshooting.
         return f"{self.response.status_code} {self.response.reason}: {self.response.text}"
+
+
+class APIJsonDecodeError(json.decoder.JSONDecodeError):
+    """
+    Error class for bad JSON responses from the REDCap API.
+
+    Includes useful details of the error in the stringification.
+    """
+    def __str__(self):
+        return f"Error: {self.msg!r} Text to decode: {self.doc!r}"
```

Comment:
I think it'd be nicer/more informative to keep the existing detailed stringification behaviour (e.g. call `super().__str__()`) but extend it with the original doc (or better yet, a length-limited substring of the original doc centered around the error position).